### PR TITLE
Added 2 New Italian Providers

### PR DESCRIPTION
**Pull request by KillerDogeEmpire**
Adippe made these providers. When I added them to my fork he told me to  see if they could be added. In StreamingCommunity Provider if you click the provider name It doesn't take you to the website, because he had to do somthing funky to get the Images. The Tantfilm provider works fine

### CODE REVIEW

**Comment by LagradOst** [comment]
I am not quite sure if I should review them as normal as you did not write these providers.

**Comment by KillerDogeEmpire** [comment]
That is fair. In discord ask Addipe how you should review them.

**Comment by KillerDogeEmpire** [comment]
image
image


Adippe told me to tell you just to review like normal and he will see what happens because I have been keeping him updated on this PR

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/StreamingcommunityProvider.kt [hunk 1]
```diff
@@ -0,0 +1,564 @@
...
+import com.lagradost.cloudstream3.LoadResponse.Companion.addTrailer
+import com.lagradost.cloudstream3.utils.*
+import org.json.JSONObject
+import java.net.URI
+import java.security.MessageDigest
+
+
+private val hlsHelper = M3u8Helper()
+
+val mapper = jacksonObjectMapper().apply {
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/StreamingcommunityProvider.kt:18]: dont use your own mapper, use parseJson instead

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/StreamingcommunityProvider.kt [hunk 2]
```diff
@@ -0,0 +1,564 @@
...
+    propertyNamingStrategy = PropertyNamingStrategy.LOWER_CAMEL_CASE
+    setSerializationInclusion(JsonInclude.Include.NON_NULL)
+}
+
+data class Moviedata (
+    @get:JsonProperty(required=true)@field:JsonProperty(required=true)
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/StreamingcommunityProvider.kt:24]: do JsonProperty("id") instead, you can look how other providers do it

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/StreamingcommunityProvider.kt [hunk 3]
```diff
@@ -0,0 +1,564 @@
...
+    val average: String,
+
+    @get:JsonProperty(required=true)@field:JsonProperty(required=true)
+    val count: Long,
+
+    @get:JsonProperty(required=true)@field:JsonProperty(required=true)
+    val type: String
+)
+
+class VideoData(elements: Collection<VideoElement>) : ArrayList<VideoElement>(elements) {
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/StreamingcommunityProvider.kt:84]: remove this

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/StreamingcommunityProvider.kt [hunk 4]
```diff
@@ -0,0 +1,564 @@
...
+    val proxyID: Long,
+
+    @get:JsonProperty(required=true)@field:JsonProperty(required=true)
+    val url: String,
+
+    @get:JsonProperty(required=true)@field:JsonProperty(required=true)
+    val type: String
+)
+
+class Seasons(elements: Collection<Season>) : ArrayList<Season>(elements) {
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/StreamingcommunityProvider.kt:124]: Remove this

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/StreamingcommunityProvider.kt [hunk 5]
```diff
@@ -0,0 +1,564 @@
...
+    companion object {
+        fun fromJson(json: String) = mapper.readValue<Seasons>(json)
+    }
+}
+
+data class Season (
+    @get:JsonProperty(required=true)@field:JsonProperty(required=true)
+    val id: Long,
+
+    val name: Any? = null,
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/StreamingcommunityProvider.kt:135]: 🤨 Any? in Json

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/StreamingcommunityProvider.kt [hunk 6]
```diff
@@ -0,0 +1,564 @@
...
+    val url: String,
+
+    @get:JsonProperty(required=true)@field:JsonProperty(required=true)
+    val type: String,
+
+    @get:JsonProperty("original_url", required=false)@field:JsonProperty("original_url", required=false)
+    val originalURL: String
+)
+
+class Trailer(elements: Collection<TrailerElement>) : ArrayList<TrailerElement>(elements) {
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/StreamingcommunityProvider.kt:198]: Remove

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/StreamingcommunityProvider.kt [hunk 7]
```diff
@@ -0,0 +1,564 @@
...
+            72 -> 3
+            73 -> 4
+            74 -> 5
+            75 -> 6
+            76 -> 7
+            77 -> 8
+            78 -> 9
+            79 -> 10
+            133 -> 11
+            else -> 0
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/StreamingcommunityProvider.kt:283]: I would return null to indicate that it is wrong

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/StreamingcommunityProvider.kt [hunk 8]
```diff
@@ -0,0 +1,564 @@
...
+            17 -> "sc-b1-02.scws-content.net"
+            18 -> "sc-b1-03.scws-content.net"
+            85 -> "sc-b1-04.scws-content.net"
+            95 -> "sc-b1-05.scws-content.net"
+            117 -> "sc-b1-06.scws-content.net"
+            141 -> "sc-b1-07.scws-content.net"
+            142 -> "sc-b1-08.scws-content.net"
+            143 -> "sc-b1-09.scws-content.net"
+            144 -> "sc-b1-10.scws-content.net"
+            else -> ""
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/StreamingcommunityProvider.kt:298]: null here as well

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/StreamingcommunityProvider.kt [hunk 9]
```diff
@@ -0,0 +1,564 @@
...
+                    datajs.releaseDate.substringBefore("-").filter { it.isDigit() }.toIntOrNull(),
+                    null,
+                )
+            }
+
+        }.toList()
+    }
+
+    override suspend fun load(url: String): LoadResponse {
+        val poster = url.substringAfter("°")
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/StreamingcommunityProvider.kt:398]: You should really not do this, the url should be the same given to you if you where using the site. IF YOU CANT get the poster in any other way then do a companion object with a map<url,posterUrl> so you can retrieve it without using the url

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/StreamingcommunityProvider.kt [hunk 10]
```diff
@@ -0,0 +1,564 @@
...
+                this.tags = datajs.genres.map { it.name }
+                this.duration = datajs.runtime?.toInt()
+                addTrailer(trailerurl)
+                this.recommendations = listacorr
+            }
+
+        }
+    }
+
+    private fun getM3u8Qualities(
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/StreamingcommunityProvider.kt:523]: Dont do this, this entire function has been replaced, check out https://github.com/LagradOst/CloudStream-3/blob/510411a9bcfe488d39691bf9a1332b292feb43af/app/src/main/java/com/lagradost/cloudstream3/animeproviders/AllAnimeProvider.kt#L320

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/TantiFilmProvider.kt [hunk 11]
```diff
@@ -0,0 +1,220 @@
...
+    )
+
+    override suspend fun getMainPage(): HomePageResponse {
+        val items = ArrayList<HomePageList>()
+        val urls = listOf(
+            Pair("$mainUrl/watch-genre/serie-tv/", "Serie Tv"),
+            Pair("$mainUrl/watch-genre/azione/", "Azione"),
+            Pair("$mainUrl/watch-genre/avventura/", "Avventura"),
+        )
+        for (i in urls) {
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/TantiFilmProvider.kt:25]: do (url, name) instead, this way you dont have to do .first and .second

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/TantiFilmProvider.kt [hunk 12]
```diff
@@ -0,0 +1,220 @@
...
+            MovieSearchResponse(
+                name,
+                href,
+                this.name,
+                TvType.Movie,
+                poster,
+                null
+            )
+
+        }.toList()
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/TantiFilmProvider.kt:68]: remove toList

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/TantiFilmProvider.kt [hunk 13]
```diff
@@ -0,0 +1,220 @@
...
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/TantiFilmProvider.kt:99]: remove toList

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/TantiFilmProvider.kt [hunk 14]
```diff
@@ -0,0 +1,220 @@
...
+                val href = element.attr("value")
+                if (season != null && season > 0 && !href.isNullOrBlank()) {
+                    list.add(Pair(season, fixUrl(href)))
+                }
+            }
+            if (list.isEmpty()) throw ErrorLoadingException("No Seasons Found")
+
+            val episodeList = ArrayList<Episode>()
+
+            for (season in list) {
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/TantiFilmProvider.kt:118]: you can do (seasonNum, seasonHref) in list instead to make it more readable

**Review by LagradOst — COMMENTED** [review]
The Json needs cleaning up and for loops can be made better if you are indexing over two values, example from kotlin tutorial: ```fun main(args: Array<String>) {
    var nums = listOf(25, 54, 68, 72)
    for((index,num) in nums.withIndex()){
        println(index.toString()+" - "+num)
    }
}```

**Pull request merged**